### ISSUE TITLE
`make test` leaves behind unused config file: ~/.minikube/profiles/config.json

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#8614]
we need to find out why we generate this usless file, I beleive maybe when there is no profile given we generate a empty config without a folder
~/.minikube/profiles/config.json

while normal profiles are in 
~/.minikube/profiles/profile_name/config.json

**Comment by afbjorklund** [comment]
I think you will find the config under ~/.minikube/profiles/minikube/config.json

**Comment by kadern0** [comment]
I've just realized that running ``make test`` generates this file.

**Comment by anencore94** [comment]
I found it was caused by the test case in `pkg/minikube/config/profile_test.go` (https://github.com/kubernetes/minikube/blob/master/pkg/minikube/config/profile_test.go#L280)

exactly the `PrimaryControlPlane(cc)` function in "old style case", 79th line in `pkg/minikube/config/proflie.go` make the converted profile and save it in `~/.minikube/profiles/config.json`, since `viper.GetString(ProfileName)` is nil.

However, I'm not sure how to fix it.

I just guess there should be nil check logic in `SaveProfile()` function.
But is there any way to get the `profileName` of such `ClusterConfig` in `PrimaryControlPlane()` function ?

I think the one who make this function wants to update the `~/.minikube/profiles/<profile_name>/config.json` file in new style format. But if it happened, the other test cases who use this config file will be polluted.

@medyagh @kadern0
